pkg/sanity/controller.go

```diff
@@ -933,3 +938,382 @@ var _ = Describe("ControllerUnpublishVolume [Controller Server]", func() {
 		Expect(err).NotTo(HaveOccurred())
 	})
 })
+
+var _ = Describe("ListSnapshots [Controller Server]", func() {
+	var (
+		c csi.ControllerClient
+	)
+
+	BeforeEach(func() {
+		c = csi.NewControllerClient(conn)
+
+		if !isControllerCapabilitySupported(c, csi.ControllerServiceCapability_RPC_LIST_SNAPSHOTS) {
+			Skip("ListSnapshots not supported")
+		}
+	})
+
+	It("should return appropriate values (no optional values added)", func() {
+		snapshots, err := c.ListSnapshots(
+			context.Background(),
+			&csi.ListSnapshotsRequest{})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(snapshots).NotTo(BeNil())
+
+		for _, snapshot := range snapshots.GetEntries() {
+			verifySnapshotInfo(snapshot.GetSnapshot())
+		}
+	})
+
+	// TODO: Add test to test for tokens
+
+	// TODO: Add test which checks list of snapshot is there when created,
+	//       and not there when deleted.
```

Comment:
I've added some tests in `ListVolumes` which are similar to what we need here. Maybe have a look and add them? https://github.com/kubernetes-csi/csi-test/pull/80
Or maybe as a follow-up PR?